plugins/guests/redhat/cap/subscription_manager.rb

```diff
@@ -18,7 +18,17 @@ def self.subscription_manager(machine)
         def self.subscription_manager_register(machine, ui)
           subscription_manager_upload_certificate(machine, ui) if machine.config.registration.ca_cert
           command = "subscription-manager register #{configuration_to_options(machine.config.registration)}"
-          machine.communicate.execute("cmd=$(#{command}); if [ \"$?\" != \"0\" ]; then echo $cmd | grep 'This system is already registered' || (echo $cmd 1>&2 && exit 1) ; fi", sudo: true)
+          error = ''
```

Comment:
We don't need to initialize the variable here. Also I would not call it error, we are just catching any output here, both from stdout and stderr.